### PR TITLE
simplify service worker gatherer and artifact format

### PR DESCRIPTION
**Pull request by brendankenny**
went in to fix a few closure warnings, but `versions` (plural) on the artifact and it being an array of either one or no values drove me a little crazy, so minor fixup to hopefully make it cleaner

### CODE REVIEW

File: src/gatherers/service-worker.js [hunk 1]
```diff
@@ -20,34 +20,46 @@ const Gather = require('./gather');
...
   reloadSetup(options) {
     const driver = options.driver;
-    this.resolved = false;
 
-    this.artifactsResolved = new Promise((res, _) => {
-      driver.on('ServiceWorker.workerVersionUpdated', data => {
-        if (!this.resolved) {
-          const controlledClients =
-              ServiceWorker.getActivatedServiceWorker(data.versions, options.url);
+    this.artifactsResolved = new Promise((resolve, reject) => {
```
Comment (paulirish) [inline src/gatherers/service-worker.js]: let's redo this thing to mimic our `getSecurityState` style.

so it'll be like `getServiceWorkerVersions`

File: src/gatherers/service-worker.js [hunk 2]
```diff
@@ -20,34 +20,46 @@ const Gather = require('./gather');
...
   }
 
+  /**
+   * @param {!Array<!ServiceWorkerVersion>} versions
+   * @param {string} url
+   * @return {(!ServiceWorkerVersion|undefined)}
+   */
   static getActivatedServiceWorker(versions, url) {
     const origin = this.getOrigin(url);
     return versions.find(v => v.status === 'activated' && v.scriptURL.startsWith(origin));
```
Comment (paulirish) [inline src/gatherers/service-worker.js]: Let's match the serviceworkermanager.js from sdk

it iterates over the results from workerVersionUpdated and extracts the `origin` for each version.
https://cs.chromium.org/chromium/src/third_party/WebKit/Source/devtools/front_end/sdk/ServiceWorkerManager.js?q=serviceworkermanager&sq=package:chromium&dr=C&l=681


Then it grabs the `origin` of the page.

and finds the one that matches (exactly).
https://cs.chromium.org/chromium/src/third_party/WebKit/Source/devtools/front_end/resources/ServiceWorkersView.js?type=cs&q=serviceworker+file:%5Esrc/third_party/WebKit/Source/devtools/front_end/&sq=package:chromium&l=72

File: src/gatherers/service-worker.js (hunk 1 above)
Comment (brendankenny) [inline src/gatherers/service-worker.js]: good call

**Comment by brendankenny** [comment]
`getSecurityState` was pretty easy, but not sure if I missed your point about the version check. Now explicitly checking the origins for equality.

**Comment by paulirish** [comment]
yup this read correctly to me. feels much better, too. :)

**Comment by paullewis** [comment]
LGTM

**Comment by paullewis** [comment]
Just a small query: are we sure that we will only ever get a single version? I imagine so, given we have a search for `activated` ones IIRC.

**Comment by paulirish** [comment]
> Just a small query: are we sure that we will only ever get a single version? I imagine so, given we have a search for activated ones IIRC.

yes it would break spec if there were >1 worker version for the origin thats activated.

**Pull request merged**